application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/GetTrialLicenseListener.java

```diff
@@ -78,46 +84,58 @@ public void onEvent(Event event, Object source, Object data)
         trialLicenseGenerator.updateLicenses();
 
         for (ExtensionId extensionId : extensions) {
-            tryGenerateTrialLicenseRecursive(extensionId);
+            InstalledExtension installedExtension = installedExtensionRepository.getInstalledExtension(extensionId);
+            Stack<ExtensionId> dependencyPath = new Stack<>();
+            dependencyPath.push(extensionId);
+            tryGenerateTrialLicenseRecursive(dependencyPath, installedExtension.getNamespaces());
         }
+
+        licensedExtensionManager.invalidateMandatoryLicensedExtensionsCache();
```

Comment:
I'm wondering if the list of mandatory licensed extensions should be invalidated before or after the trial license generation (which needs this list). ATM it seems the trial license generation is using the old list (before the install job). Is this OK?